src/main/java/org/quantumbadger/redreader/reddit/CommentListingRequest.java

```diff
@@ -296,6 +302,43 @@ private void buildCommentTree(
 		} else if(thing.getKind() == RedditThing.Kind.COMMENT) {
 
 			final RedditComment comment = thing.asComment();
+
+			if (comment.media_metadata != null && comment.media_metadata.asObject() != null) {
+				try {
+					final JsonObject metadataObj = comment.media_metadata.asObject();
+
+					final Iterator<Map.Entry<String, JsonValue>> jsonIterator = metadataObj
+							.iterator();
+
+					while (jsonIterator.hasNext()) {
+						final String key = jsonIterator.next().getKey();
+
+						final EmoteMetadata emoteMetadata = metadataObj
+								.get(key).asObject()
+								.asObject(EmoteMetadata.class);
+
+						// id is always structured as emote|{subreddit_id}|{emote_id}
+						// for subreddit emotes
+						if (emoteMetadata.id.split("\\|")[0].equalsIgnoreCase("emote")) {
+							final String emoteId = emoteMetadata.id.split("\\|")[2];
+
+							final String emotePlaceholder = String.format(Locale.getDefault(),
+									":%s:", emoteId);
+
+							final String imgTag = String.format(Locale.getDefault(),
+									"<img src=\"%s\" title=\"%s\"></img>",
+									emoteMetadata.s.u,
+									emotePlaceholder);
+
+							comment.body_html = comment.body_html.replace(emotePlaceholder, imgTag);
+						}
+					}
+				} catch (final Exception e) {
```

Comment:
We shouldn't ignore exceptions -- at the very least this should be logged, although typically we'd want to show a parse error next to the relevant comment so that the user can report this.

Reply:
Makes sense. I'll make it show a parse error to the user

Reply:
I'm logging this exception now. 

I don't think I've ever seen an exception be thrown here so I didn't bother with showing a parse error for this. I'll look into it if you think it's important